src/core/reactor.ts

```diff
@@ -983,25 +983,57 @@ protected _getFirstReactionOrMutation(): Reaction<any> | undefined {
             }
 
             // Add the new edge.
+            // Add edge when we are able to add graph
+            // if(isAbleToAdd(dst, src)) graph.addEdge(dst, src)
             graph.addEdge(dst, src)
 
             // 1) check for loops
+            let hasCycle = false
             if (graph.hasCycle()) {
-                return false
+                hasCycle = true
+                // throw Error("New connection introduces cycle.")
             }
 
             // 2) check for direct feed through.
+            let hasDirectFeedThrough = false
             let inputs = this._findOwnInputs()
-            for (let output of this._findOwnOutputs()) {
+            let outputs = this._findOwnOutputs()
+
+
+            for (let output of outputs) {
                 let newReachable = graph.reachableOrigins(output, inputs)
                 let oldReachable = this._causalityGraph.reachableOrigins(output, inputs)
 
+                // Print newReachable and oldReachable origin
+                for (let new_ori of newReachable) {
+                    if (new_ori instanceof Port) {
+                        console.log("New Origin: ", new_ori._getFullyQualifiedName())
+                    }              
+                }
+                for (let old_ori of oldReachable) {
+                    if (old_ori instanceof Port) {
+                        console.log("Old Origin: ", old_ori._getFullyQualifiedName())
+                    }              
+                }
+                console.log("!!!")    
+
                 for (let origin of newReachable) {
                     if (origin instanceof Port && !oldReachable.has(origin)) {
-                        return false
+                        hasDirectFeedThrough = true
+                        // throw Error("New connection introduces direct feed through.")
```

Comment:
Nit: should be deleted?

Reply:
I think the reachable origin function is not entirely implemented. So, we need to remain this part.

Reply:
Ah, I meant this line:

// throw Error("New connection introduces direct feed through.")

Reply:
Yes! It looks not needed. It was reflected in https://github.com/lf-lang/reactor-ts/pull/108/commits/1f5caf8b429157539288ee67e3a49eeb0e63b3db